homeassistant/components/sensor/pushbullet.py

```diff
@@ -0,0 +1,120 @@
+"""
+PushBullet platform for sensor component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.pushbullet/
+"""
+import logging
+import threading
+import voluptuous as vol
+
+from pushbullet import PushBullet
+from pushbullet import InvalidKeyError
+from pushbullet import Listener
+
+from homeassistant.const import (CONF_API_KEY, CONF_MONITORED_CONDITIONS)
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pushbullet.py==0.10.0']
+
+SENSOR_TYPES = {
+    'application_name': ['Application name'],
+    'body': ['Body'],
+    'notification_id': ['Notification ID'],
+    'notification_tag': ['Notification tag'],
+    'package_name': ['Package name'],
+    'receiver_email': ['Receiver email'],
+    'sender_email': ['Sender email'],
+    'source_device_iden': ['Sender device ID'],
+    'title': ['Title'],
+    'type': ['Type']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(
+        CONF_MONITORED_CONDITIONS,
+        default=['title', 'body']): vol.All(
+            cv.ensure_list, vol.Length(min=1),
+            [vol.In(SENSOR_TYPES.keys())]
+            ),
+    vol.Required(CONF_API_KEY): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    # Get the PushBullet notification service.
+    try:
+        pushbullet = PushBullet(config.get(CONF_API_KEY))
+    except InvalidKeyError:
+        _LOGGER.error(
+            "Wrong API key supplied.{" + config.get(CONF_API_KEY) + "}"
```

Comment:
I think that we should not print the API key to the log. The key is wrong, period. It could be that the given API key is valid in a different context, thus better not store it in the log.